packages/api/cms-api/src/translation/translation.module.ts

```diff
@@ -0,0 +1,26 @@
+import { DynamicModule, Global, Module } from "@nestjs/common";
+
+import { TranslationServiceInterface } from "./translation.config";
+import { TranslationResolver } from "./translation.resolver";
+import { TRANSLATION_CONFIG } from "./translation-config.constants";
+
+interface TranslationConfig {
+    service: TranslationServiceInterface;
+}
+
+@Global()
+@Module({})
+export class TranslationModule {
+    static register(config: TranslationConfig): DynamicModule {
```

Comment:
This needs to support asynchronous registration. The translation service in the application shouldn't be provided by this module, but by a module in the application. For instance, there could be a `DeepLModule` that provides the service. This translation module would then need to import the module and use the service that's exported there.

Reply:
Resolved outdated.